src/views/RecIM/components/List/Listing/index.jsx

```diff
@@ -378,21 +404,38 @@ const ParticipantListing = ({
         }
         disablePadding
       >
-        <ListItemButton
-          to={`/profile/${participant.Username}`}
-          className={`${styles.listing} ${
-            withAttendance && (didAttend ? styles.attendedListing : styles.absentListing)
-          }`}
-        >
-          <ListItemAvatar>
-            <Avatar
-              src={`data:image/jpg;base64,${avatar}`}
-              className={minimal ? styles.avatarSmall : styles.avatar}
-              variant="rounded"
-            ></Avatar>
-          </ListItemAvatar>
-          <ListItemText primary={name} secondary={participant.Role} />
-        </ListItemButton>
+        {participant.IsCustom ? (
+          <ListItem
+            className={`${styles.listing} ${
+              withAttendance && (didAttend ? styles.attendedListing : styles.absentListing)
+            }`}
+          >
+            <ListItemAvatar>
+              <Avatar
+                src={`data:image/jpg;base64,${avatar}`}
+                className={minimal ? styles.avatarSmall : styles.avatar}
+                variant="rounded"
+              ></Avatar>
+            </ListItemAvatar>
+            <ListItemText primary={name} secondary={participant.Role} />
+          </ListItem>
+        ) : (
+          <ListItemButton
+            to={`/profile/${participant.Username}`}
+            className={`${styles.listing} ${
+              withAttendance && (didAttend ? styles.attendedListing : styles.absentListing)
+            }`}
+          >
+            <ListItemAvatar>
+              <Avatar
+                src={`data:image/jpg;base64,${avatar}`}
+                className={minimal ? styles.avatarSmall : styles.avatar}
+                variant="rounded"
+              ></Avatar>
+            </ListItemAvatar>
+            <ListItemText primary={name} secondary={participant.Role} />
+          </ListItemButton>
+        )}
```

Comment:
This chunk of code seems odd. Why does a custom participant not need the `<ListItemButton>`. Also, if the nesting is required, can you just make a helper function that wraps the sub component? It's the exact same piece of code.